daemon/src/setup_contract.rs

```diff
@@ -315,11 +341,13 @@ pub async fn new(
             })
             .collect::<Result<HashMap<_, _>>>()
     }})
+    .instrument(tracing::debug_span!("Do some work on CETs")) // TODO(restioson) what is this doing?
```

Comment:
Congratulations, you found the potentially ugliest part of our code-base 😬

In here we are extracting the counterparty's adaptor signature from their message to us.